### DISCUSSION TITLE
Deleting a 2D slice in a 3D stack

### DISCUSSION THREAD

**Discussion by ckolluru** [zarr-developers/community#63] [General]
Hi,

I have a Zarr dataset of a 3D volume in ZYX format. If the total dataset size is (z, y, x) pixels, the chunk size is set to (16, y, x).

After the dataset is created, I would need to manually verify the slices and remove slices with artifacts. 

Is there a way to do this in zarr and if not, would it be beneficial to implement?

Thanks,
Chaitanya

**Comment by rabernat** [comment]
It depends on what you mean by "remove".

It is possible to delete the chunks and keep the array size the same--the deleted chunks would be replaced with the array fill value.

It is not currently possible to delete chunks and resize the array to become smaller.

**Reply by ckolluru** [reply]
Thanks for your response.

I was thinking of deleting a single 2D slice within a specific chunk of size (16, y, x). I would want the array size to be reduced by 1 unit in z.

Good to know that it is not currently supported. So the only way right now would be to create a new dataset entirely?

**Reply by rabernat** [reply]
There are various workarounds you could consider. However, given your chunk size, removing a single slice of shape (1, y, x) would result in uneven chunks. That is also not supported, although it has been proposed--see https://zarr.dev/zeps/draft/ZEP0003.html.
